monad-eth-testutil/examples/txgen/workers/rpc_request_gen.rs

```diff
@@ -346,7 +739,54 @@ impl RpcWalletSpam {
         }
     }
 
-    pub async fn run(&self, shutdown: Arc<AtomicBool>) {
+    pub async fn run_indexer_workflow(&self, shutdown: Arc<AtomicBool>) {
+        // start block tip refresher task
+        let (tip_sender, mut tip_receiver) = tokio::sync::mpsc::channel(8);
+        let (index_done_sender, mut index_done_receiver) = tokio::sync::mpsc::channel(32);
+        let refresher = TipRefresher::new(self.rpc_client.clone(), tip_sender);
+        tokio::spawn(async move { refresher.run().await });
+
+        let mut status_interval = tokio::time::interval(Duration::from_secs(1));
+        status_interval.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Delay);
+
+        let mut total_indexed = 0;
+        let mut total_failed = 0;
+        let mut uniq_addrs = BTreeSet::new();
+
+        // select on tip refresher channel and completion channel
+        while !shutdown.load(Ordering::Relaxed) {
+            tokio::select! {
+                Some((index_from, index_to)) = tip_receiver.recv() => {
+                    debug!(from=index_from, to=index_to, "index range");
+
+                    for block_number in index_from..=index_to {
+                        RpcRequestGenerator::index_block(self.rpc_client.clone(), block_number, self.requests_per_block, index_done_sender.clone()).await;
```

Comment:
If you are doing these calls synchronously, then why are you sending the result back on a channel instead of returning them from the fn? 
My guess is you meant for these to run async? Since there's a decent chance doing this sync could cause the indexer to fall behind.